src/libstd/sync/mpsc/mod.rs

```diff
@@ -336,51 +338,123 @@ unsafe impl<T: Send> Send for Receiver<T> { }
 #[stable(feature = "rust1", since = "1.0.0")]
 impl<T> !Sync for Receiver<T> { }
 
-/// An iterator over messages on a receiver, this iterator will block whenever
-/// [`next`] is called, waiting for a new message, and [`None`] will be returned
+/// An iterator over messages on a [`Receiver`], created by [`iter`].
+/// This iterator will block whenever [`next`] is called,
```

Comment:
Can you put a newline after the first sentence? The summary is quite long otherwise.